bam_consensus.c

```diff
@@ -468,63 +958,69 @@ int nm_init(void *client_data, samFile *fp, sam_hdr_t *h, pileup_t *p) {
         return -1;
     p->cd = local_nm;
 
+    double poly_adj = opts->homopoly_fix ? opts->homopoly_fix : 1;
+
     if (opts->adj_qual) {
-#if 0
-        // Tweak by localised quality.
-        // Quality is reduced by a significant portion of the minimum quality
-        // in neighbouring bases, on the pretext that if the region is bad, then
-        // this base is bad even if it claims otherwise.
+        // Skew local NM by qual vs min-qual delta
         uint8_t *qual = bam_get_qual(b);
-        const int qhalo = 8; // 2?
-        int qmin = 50; // effectively caps PacBio qual too
+        uint8_t *seq = bam_get_seq(b);
+        const int qhalo = 8; // 4
+        int qmin = 99, qminp = 99;
+        int base = bam_seqi(seq, 0), polyl = 0, polyr=0; // pos, not len
+        const int qhalop = 2;
+
+        for (i = 1; i < qlen; i++) {
+            if (bam_seqi(seq, i) != base)
+                break;
+            if (i < qhalop && qminp > qual[i])
+                qminp = qual[i];
+        }
+        polyr = i-1;
```

Comment:
Only if `opts->mode == MODE_BAYES_116`.

We assign `int pl = polyr-polyl` (homopolymer length), which then gets used in the calculation of `t`.  That's the minimum quality threshold such that lower values cause us to start aggregating `local_nm` qual adjustments, as evident in the `t < qual[i] ? qual[i]-t : 0` below.

Arguably some of the code could be behind conditionals to avoid computation when options mean the results won't be used, but I doubt it'll help much in general usage.

Reply:
Sorry I see what you mean now.  Polyr is used, but it's reassigned again after this assignment.  I'll see what can be culled.  The initial assignment of polyl is needed however.

It's possible qminp can be culled too from that initial loop, with a small change.